src/main/java/jewellery/inventory/mapper/SaleMapper.java

```diff
@@ -0,0 +1,85 @@
+package jewellery.inventory.mapper;
+
+import java.time.Instant;
+import java.util.*;
+import jewellery.inventory.dto.request.SaleRequestDto;
+import jewellery.inventory.dto.response.ProductResponseDto;
+import jewellery.inventory.dto.response.SaleResponseDto;
+import jewellery.inventory.model.Product;
+import jewellery.inventory.model.Sale;
+import jewellery.inventory.model.User;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class SaleMapper {
+  private final UserMapper userMapper;
+  private final ProductMapper productMapper;
+
+  public SaleResponseDto mapEntityToResponseDto(Sale sale) {
+    SaleResponseDto saleResponseDto = new SaleResponseDto();
+    saleResponseDto.setSeller(userMapper.toUserResponse(sale.getSeller()));
+    saleResponseDto.setBuyer(userMapper.toUserResponse(sale.getBuyer()));
+    saleResponseDto.setProducts(mapAllProductsToResponse(sale));
+    saleResponseDto.setTotalPrice(getTotalPriceFromEntity(sale.getProducts()));
+    saleResponseDto.setTotalDiscount(calculateDiscount(sale.getProducts(), "amount"));
+    saleResponseDto.setTotalDiscountedPrice(calculateDiscount(sale.getProducts(), "percentage"));
```

Comment:
Please, extract "amount" and "percentage" into constants and use them instead of the hardcoded strings.